pkg/flow/adapter/xmltojsontransformation/adapter.go

```diff
@@ -104,9 +104,20 @@ func (a *Adapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloud
 		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
 	}
 
+	if err := a.emitToSink(ctx, event); err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
+	}
+
 	return &event, cloudevents.ResultACK
```

Comment:
Do I understand correctly that we both send the event to the sink _and_ respond with the event?

If yes, we probably want to short circuit the last `return` in case a sink was provided (I'm assuming the sink parameter is optional, not sure what the intention was?)

Reply:
hey just saw this! This was the main reason i pulled this back as a draft. I wanted to discuss how to handle this in the eng. meeting this morning with @cab105 and @odacremolbap .. I have now updated it to respond if no sink is specified.

Reply:
Looking at the reconciler logic, the controller will fail to bring up the adapter if `K_SINK` is not defined so unless it really is optional, then it is safe that `K_SINK` will always be present.

Reply:
I have made updates to address this :)